src/1_api.js

```diff
@@ -51,12 +51,32 @@ Server.prototype.serializeObject = function(obj, prefix) {
  * @param {Object.<string, *>} data
  */
 Server.prototype.getUrl = function(resource, data) {
-	var k, v, err;
-	var url = resource.destination + resource.endpoint;
+	var k, v, err,
```

Comment:
yep, I agree. I can do the refactory later on in a separate PR. In the meantime, allow me to leave this comment here kind of like a todo for myself. Let me know if you think otherwise :)